slaves/linux-cross/README.md

```diff
@@ -0,0 +1,102 @@
+
+To cross compile libstd/rustc we need a C cross toolchain: a cross gcc and a cross compiled libc.
+For some targets, we use crosstool-ng to build these toolchains ourselves instead of using the ones
+packaged for Ubuntu because:
+
+- We can control the glibc version of the toolchain. In particular, we can lower its version as much
+    as possible, this lets us generate libstd/rustc binaries that run in systems with old glibcs.
+- We can create toolchains for targets that don't have an equivalent package available in Ubuntu.
+
+crosstool-ng uses a `.config` file, generated via a menuconfig interface, to specify the target,
+glibc version, etc. of the toolchain to build. Because this menuconfig interface requires user
+intervention we store pre-generated `.config` files in this repository to keep the `docker build`
+command free of user intervention.
+
+The next section explains how to generate a `.config` file for a new target, and the one after that
+contains the changes, on top of the default toolchain configuration, used to generate the `.config`
+files stored in this repository.
+
+## Generating a `.config` file
+
+If you have a `linux-cross` image lying around you can use that and skip the next two steps.
+
+- First we spin up a container and copy `build_toolchain_root.sh` into it. All these steps are
+    outside the container:
+
+```
+# Note: We use ubuntu:15.10 because that's the "base" of linux-cross Docker image
+$ docker run -it ubuntu:15.10 bash
+$ docker ps
+CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
+cfbec05ed730        ubuntu:15.10        "bash"              16 seconds ago      Up 15 seconds                           drunk_murdock
+$ docker cp build_toolchain_root.sh drunk_murdock:/
+```
+
+- Then inside the container we build crosstool-ng by simply calling the bash script we copied in the
+    previous step:
+
+```
+$ bash build_toolchain_root.sh
+```
+
+- Now, inside the container run the following command to configure the toolchain. To get a clue of
+    which options need to be changed check the next section and come back.
+
+```
+$ ct-ng menuconfig
+```
+
+- Finally, we retrieve the `.config` file from the container and give it a meaningful name. This is
+    done outside the container.
+
+```
+$ docker drunk_murdock:/.config arm-linux-gnueabi.config
+```
+
+- Now you can shutdown the container or repeat the two last steps to generate a new `.config` file.
+
+## Toolchain configuration
+
+Changes on top of the default toolchain configuration used to generate the `.config` files in this
+directory. The changes are formatted as follows:
+
+```
+$category > $option = $value -- $comment
+```
+
+## `arm-linux-gnueabi.config`
+
+For targets: `arm-unknown-linux-gnueabi`
+
+- Path and misc options > Prefix directory = /x-tools/${CT_TARGET}
+- Target options > Target Architecture = arm
+- Target options > Architecture level = armv5t -- (*)
```

Comment:
> Although now that I think about it I'm not sure it matters all that much. This largely only affects glibc, and we're not distributing glibc, so...

I thought you concluded, from this comment, that this default architecture level doesn't affect the C libraries we build, apart from glibc, because we always pass the `-march` flag to gcc when compiling them. But, perhaps, my guess about your conclusion was wrong!

I'm OK with changing both to armv6 though.